libminifi/src/FlowController.cpp

```diff
@@ -438,6 +440,20 @@ std::shared_ptr<state::response::ResponseNode> FlowController::getAgentManifest(
   return agentInfo;
 }
 
+void FlowController::executeOnAllComponents(std::function<void(state::StateController*)> func) {
+  std::lock_guard<std::recursive_mutex> lock(mutex_);
+  for (auto* component: getAllComponents()) {
+    func(component);
+  }
+}
+
+void FlowController::executeOnComponent(const std::string &name, std::function<void(state::StateController*)> func) {
+  std::lock_guard<std::recursive_mutex> lock(mutex_);
+  if (auto* component = getComponent(name); component != nullptr) {
+    func(component);
+  }
```

Comment:
maybe add an error log if the component is not found?  I think that should never happen, and a log could be useful if it does

Reply:
Done.

Reply:
You can even leave out the second part, and it means the same.
```
if (auto* const component = getComponent(name)) {
  func(component);
}
```